### PR TITLE
Add credentials for QR codes

### PR DESCRIPTION
**Pull request by ruioliveira02**
I tried doing something more generic by making a credential available to every single identity role in the system

Missing a few things:
- Blocking routes by role / user (what do we want here? Mentors can edit everything, limit to the own user (guardian in case of ninja))
- Script to generate them in production
- Script to generate the physical credentials from a base image
- Tests

### CODE REVIEW

File: lib/bokken/accounts/credential.ex [hunk 1]
```diff
@@ -0,0 +1,51 @@
+defmodule Bokken.Accounts.Credential do
+  @moduledoc """
+  A credential (QR Code) the uniquely identifies someone
```
Comment (ruilopesm) [inline lib/bokken/accounts/credential.ex:3]:
```suggestion
  A credential (QR Code) that uniquely identifies someone
```

File: lib/bokken/accounts/credential.ex [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    case changeset.valid? do
+      true ->
+        set_fields = Enum.filter(@fkeys, fn field -> get_field(changeset, field) != nil end)
+
+        if length(set_fields) <= 1 do
+          changeset
+        else
+          add_error(
+            changeset,
+            List.first(setFields),
```
Comment (ruilopesm) [inline lib/bokken/accounts/credential.ex]:
```suggestion
            List.first(set_fields),
```

File: lib/bokken_web/views/credentials_view.ex [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  def render("error.json", %{reason: reason}) do
+    %{reason: reason}
+  end
+
+  def render("credential.json", %{credential: credential}) do
+    case credential do
+      %{id: id, ninja_id: ninja_id} -> %{id: id, ninja_id: ninja_id}
+      %{id: id, mentor_id: mentor_id} -> %{id: id, mentor_id: mentor_id}
+      %{id: id, organizer_id: organizer_id} -> %{id: id, organizer_id: organizer_id}
+      %{id: id, guardian_id: guardian_id} -> %{id: id, guardian_id: guardian_id}
```
Comment (ruilopesm) [inline lib/bokken_web/views/credentials_view.ex]: Isn't this the same of literally rendering the credential itself?

Comment (ruioliveira02) [reply lib/bokken_web/views/credentials_view.ex]: No has there can be more parameters that we don't want to publish (like timestamps)

**Comment by ruilopesm** [comment]
Feature not being implemented.

**Pull request closed without merging**